libs/filterData.js

```diff
@@ -26,17 +26,35 @@ module.exports = (entries, doctype, options = {}) => {
   //  - keys : this is the list of keys used to check that two items are the same
   const keys = options.keys ? options.keys : ['_id']
   log('debug', keys, 'keys')
-  const index = options.index ? options.index : cozy.data.defineIndex(doctype, keys)
-  return index
-  .then(i => {
+
+  const createHash = item => {
+    return keys.map(key => {
+      let result = item[key]
+      if (key === 'date') result = new Date(result).valueOf()
```

Comment:
why a different case for "date" ? What if there are other dates ?

Reply:
Ah mais c'est pas toi qui l'a écris ca ...

Reply:
J'ai déplacé le code https://github.com/cozy/cozy-konnector-libs/pull/64/files#diff-d4128108c657a5bdfe0178e0e7d6f8a0L59

J'ai ajouté le `.valueOf()` pour avoir un entier qui selon moi est mieux pour générer un hash. Après est ce que ça fonctionne si on supprime le if je vais regarder.

Reply:
@doubleface tu te souviens du pourquoi ?

Reply:
> qui selon moi est mieux

Pourquoi ?

Et pis c'est pas vraiment un hash ce truc c'est plutôt une clé en fait.

Reply:
En fait, le code n'est pas de moi non plus, ça vient de konnectors v2, en tous cas pour cette partie.